crates/storage_api2/src/testing.rs

```diff
@@ -88,6 +96,107 @@ impl StorageApi for FakeStorage {
             operation::Owned::HScan(op) => this.hscan(op).into(),
         })
     }
+
+    async fn pull_data(
+        &self,
+        keyrange: RangeInclusive<u64>,
+        _keyspace_version: KeyspaceVersion,
+    ) -> Result<impl Stream<Item = Result<DataItem>> + Send> {
+        let this = self.inner.lock().unwrap();
+
+        if this.broken {
+            return Err(Error::new(ErrorKind::Internal));
+        }
+
+        let start = keyrange.start().to_be_bytes().to_vec();
```

Comment:
worth commenting somewhere we're  relying big endianness so we dont mess up later when doing some byte manipulation